app/webpack/javascripts/modules/determination.mjs

```diff
@@ -0,0 +1,77 @@
+/**
+ * Determination
+ *
+ * Calculate totals for the assessment summary on case_workers/claims/<claim-id>
+ */
+export class Determination {
+  /**
+   * @param {Element} $module - HTML element to use for component
+   */
+  constructor ($module) {
+    if (($module instanceof window.HTMLElement) && document.body.classList.contains('govuk-frontend-supported')) {
+      this.$module = $module
+    }
+  }
+
+  /**
+   * Initialise component
+   */
+  init () {
+    // Check that required elements are present
+    if (!this.$module) {
+      return
+    }
+
+    const $module = this.$module
+
+    this.$totalExclVat = document.getElementsByClassName('js-total-exc-vat-determination')[0]
+    this.$totalVat = $module.getElementsByClassName('js-vat-determination')[0]
+    this.$LgfsVat = $module.getElementsByClassName('js-lgfs-vat-determination')[0]
+    this.$totalInclVat = $module.getElementsByClassName('js-total-determination')[0]
+    this.scheme = $module.dataset.scheme
+    // Should this be this.applyVat?
+    this.ajaxVat = $module.dataset.applyVat
+    this.vatUrl = $module.dataset.vatUrl
+    this.vatDate = $module.dataset.submittedDate
+
+    this.fields = [
+      document.getElementById('claim_assessment_attributes_fees'),
+      document.getElementById('claim_assessment_attributes_expenses'),
+      document.getElementById('claim_assessment_attributes_disbursements'),
+      document.getElementById('claim_assessment_attributes_vat_amount')
+    ].filter(field => field)
+    this.fields.forEach(element => element.addEventListener('change', () => this.calculateTotalRows()))
+  }
+
+  calculateTotalRows () {
+    const total = this.fields.reduce((n, field) => n + this.parsedValue(field), 0).toFixed(2)
+
+    this.applyVat(total).then(data => {
+      this.$totalExclVat.innerHTML = data.net_amount
+      if (this.$totalVat) { this.$totalVat.innerHTML = data.vat_amount }
+      this.$totalInclVat.innerHTML = data.total_inc_vat
+    })
+  }
+
```

Comment:
You can use async await here. Makes `applyVat` to return a promise that can be awaited. Also if you are using `parsedFloat` only here that also can be removed/ or not ! 
```javascript
async function calculateTotalRows() {
  const total = this.fields.reduce((n, field) => n + (parseFloat(field?.value) || 0.0), 0).toFixed(2);
  const { net_amount, vat_amount, total_inc_vat } = await this.applyVat(total);
  this.$totalExclVat.innerHTML = net_amount;
  if (this.$totalVat) { this.$totalVat.innerHTML = vat_amount; }
  this.$totalInclVat.innerHTML = total_inc_vat;
}
```

Reply:
I tried making this into an async function and now SonarCloud is flagging a bug at the point at which it is being called: "Promise returned in function argument where a void return was expected." I remember when I tried this first I was getting similar issues and couldn't work out how to resolve it so I opted for it not being async. Is there a way to fix this?